classes/context/ContextDAO.inc.php

```diff
@@ -144,10 +144,9 @@ function getBySetting($settingName, $settingValue, $contextId = null) {
 	function resequence() {
 		$result = $this->retrieve('SELECT ' . $this->primaryKeyColumn . ' AS context_id FROM ' . $this->tableName . ' ORDER BY seq');
 		$i=1;
-		for ($i=1; $row = (array) $result->current(); $i+=2 && $result->next()) {
+		for ($i = 1; $row = (array) $result->current(); $i += 1 && $result->next()) {
```

Comment:
As you're touching this old code:
- I don't see a reason to convert the object into an array.
- The `1 && $result->next()` turns into a `boolean`, I think it would be better to separate the statements.
  - With a comma: `for ($i = 1; $row = $result->current(); ++$i, $result->next()) { blabla; }`
  - Or to move it to another line : `for ($i = 1; $row = $result->current(); ++$i) { blabla; $result->next(); }

Reply:
Updated to use `foreach`